src/site/pages/digital/tests/containers/ExprToCircuitPopup.test.tsx

```diff
@@ -0,0 +1,143 @@
+import "@testing-library/jest-dom";
+import {act, render, screen}          from "@testing-library/react";
+import userEvent                      from "@testing-library/user-event";
+import {Provider}                     from "react-redux";
+import {applyMiddleware, createStore} from "redux";
+import thunk, {ThunkMiddleware}       from "redux-thunk";
+
+import {Setup} from "test/helpers/Setup";
+
+import {LED, ORGate, Switch} from "digital/models/ioobjects";
+
+import {OpenHeaderPopup} from "shared/state/Header";
+
+import {GetToggles} from "shared/tests/helpers/GetToggles";
+
+import {AppState} from "site/digital/state";
+
+import {AllActions} from "site/digital/state/actions";
+import {reducers}   from "site/digital/state/reducers";
+
+import {ExprToCircuitPopup} from "site/digital/containers/ExprToCircuitPopup";
+
+
+// beforeAll and beforeEach can be used to avoid duplicating store/render code, but is not recommended
+//  see: https://testing-library.com/docs/user-event/intro
+describe("Main Popup", () => {
+    const info = Setup();
+    const store = createStore(reducers, applyMiddleware(thunk as ThunkMiddleware<AppState, AllActions>));
+    const user = userEvent.setup();
+
+    beforeEach(() => {
+        render(<Provider store={store}><ExprToCircuitPopup mainInfo={info} /></Provider>);
```

Comment:
Does `render` completely reset and start with only this? or does it add to the previous `render` call? i.e. are there a bunch of these virtually being added before each test?

Reply:
The library does some automatic cleanup between tests and basically derenders everything, so it is necessary to do this before each test

Reply:
Okay cool